sympy/ntheory/factor_.py

```diff
@@ -311,6 +315,59 @@ def multiplicity(p, n):
     return m
 
 
+def multiplicity_in_factorial(b, n):
+    """
+    Finds the greatest integer m such that b**m divides factorial of n without calculating the factorial
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory import multiplicity_in_factorial
+
+    >>> multiplicity_in_factorial(5, 60)
+    14
+    >>> multiplicity_in_factorial(5, 25)
+    6
+    >>> multiplicity_in_factorial(12, 49)
+    22
+    >>> multiplicity_in_factorial(10, 100)
+    24
+    >>> multiplicity_in_factorial(42, 10**10)
+    1666666660
+
+    """
+
+    b = as_int(b)
+    n = as_int(n)
+
+    if b <= 0:
+        raise ValueError('expecting positive integer got %s' % b )
+
+    if n < 0:
+        raise ValueError('expecting non-negative integer got %s' % n )
+
+    m = 0
+
+    primelist = factorint(b)
+
+    for p in primelist:
+        multiples_curr_sum = 0
+        curr_factor = p
+        curr_multiple = n
+
+        while curr_multiple != 0:
+            curr_multiple = n//curr_factor
```

Comment:
This is just the digit of n in base n -- that's why I suggested the few line equivalent for this. Without explanation someone has to digest this routine to see what it is that is being done.

Reply:
so are the suggested changes then?

Reply:
Done